Update CLI documentation for status and reset commands (resolves #4771)

Wryhder opened the pull request:
<!-- Thank you for sending a pull request :heart: -->

## Current behavior

<!-- Please describe the current behavior of the code before the changes in this pull request are applied. -->

The docs for the `status` and `reset` commands are missing code examples and a long about section.

## Proposed changes

<!-- Please describe the changes proposed in this pull request. -->
<!-- If this pull request resolves an already recorded bug or a feature request, please add a link to that issue. -->

Resolves #4771 

The modifications are as suggested in [this comment](https://github.com/build-trust/ockam/issues/4771#issuecomment-1563913472).

## Checks

<!--
To help us review and merge this pull request quickly, please confirm the following
by replacing the [ ] in front of each bullet point below with [x]
-->

- [x] All commits in this Pull Request are [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) and Verified by Github.
- [x] All commits in this Pull Request follow the Ockam [commit message convention](https://github.com/build-trust/.github/blob/main/CONTRIBUTING.md#commit-messages).
- [x] There are no Merge commits in this Pull Request. Ockam repo maintains a linear commit history. We merge Pull Requests by rebasing them onto the develop branch. Rebasing to the latest develop branch and force pushing to your Pull Request branch is okay.
- [x] I have read and accept the Ockam Community [Code of Conduct](https://github.com/build-trust/.github/blob/main/CODE_OF_CONDUCT.md).
- [x] I have read and accepted the Ockam [Contributor License Agreement](https://github.com/build-trust/.github/blob/main/CLA.md) by adding my Git/Github details in a row at the end of the [CONTRIBUTORS.csv](https://github.com/build-trust/ockam/blob/develop/.github/CONTRIBUTORS.csv) file in a separate pull request to the [build-trust/ockam](https://github.com/build-trust/ockam) repository. The easiest way to do this is to [edit the CONTRIBUTORS.csv](https://github.com/build-trust/ockam/edit/develop/.github/CONTRIBUTORS.csv) file in the github web UI and create a separate Pull Request, this will mark the commit as verified.

<!-- We're looking forward to merging your contribution!! -->

Wryhder commented:
@adrianbenavides Hi! Could you please address these?

1. ~~Is this expected behaviour? Some information isn't included in the output of `ockam status` until after some other command is executed. For example, I ran `ockam message send hello --to /node/n1/service/uppercase` successfully after starting the node, but only after running `ockam service list --at n1` did the Transport, Secure Channel, and Services information populate. Also, only after running `ockam space list` was subscription info included in the status output.~~
2. ~~Please help with any needed clarifications for [this comment](https://github.com/build-trust/ockam/issues/4771#issuecomment-2267510477).~~
3. Please review the [long about](https://github.com/build-trust/ockam/pull/8070/commits/6199b9221777391633b080a036908421bdaff4d7#diff-e3ee3eff2828dd84252702e52b5b96e7ab705e2129a6278d767992926177fb1f)  for status in my commit. Is that useful information or unnecessary duplication? Should anything in particular be elaborated on, and in what way?
4. When might it be useful or necessary to do a reset (asides maybe discarding after initially testing or trying things out)?

adrianbenavides commented on implementations/rust/ockam/ockam_command/src/reset/static/after_long_help.txt:
```diff
@@ -0,0 +1,6 @@
+# Removes the local Ockam configuration including all Spaces, Identities, and Nodes
```
One note here. By default, the command will only remove the local state, basically the `database.sqlite3` file at `$OCKAM_HOME`. 

If you pass the `--all` argument, it will also try deleting the orchestrator resources (i.e. spaces and projects).

adrianbenavides commented on implementations/rust/ockam/ockam_command/src/reset/static/after_long_help.txt:
```diff
@@ -0,0 +1,6 @@
+$ ockam reset
+
+# To confirm without prompting
+$ ockam reset -y
```
I'll also add a third example with the `--all` flag

adrianbenavides commented on implementations/rust/ockam/ockam_command/src/status/static/long_about.txt:
```diff
@@ -0,0 +1,8 @@
+- all cryptographic identities stored in your vault and their enrollment status with the Ockam Orchestrator
+- the status of any Nodes, associated routes and protocols, listeners and services, and more
+
+  NB: To see the list of workers running in a node, run `ockam worker list --at <NAME_OF_NODE>`
```
This last line can be omitted. Workers are a lower-level primitive that most users won't need to interact with directly.

adrianbenavides commented on implementations/rust/ockam/ockam_command/src/status/static/long_about.txt:
```diff
@@ -0,0 +1,8 @@
+- the controller (Orchestrator) and project version
+- your spaces and the users associated with them
+- all cryptographic identities stored in your vault and their enrollment status with the Ockam Orchestrator
+- the status of any Nodes, associated routes and protocols, listeners and services, and more
```
Let's replace:
- controller (Orchestrator) -> Ockam Orchestrator
- project, space, identity -> capitalized


Also, the list won't look good on the terminal (when running `ockam status --help`). So let's try to summarize the information in a paragraph.

adrianbenavides commented on implementations/rust/ockam/ockam_command/src/status/mod.rs:
```diff
@@ -22,8 +22,15 @@ use ockam_api::nodes::{BackgroundNodeClient, InMemoryNode};
+
 /// Display information about the system's status
 #[derive(Clone, Debug, Args)]
+#[command(
```
Perhaps it would be better to replace "Display information about the system's status" by "Display information about your Ockam instance"

adrianbenavides commented on implementations/rust/ockam/ockam_command/src/status/static/after_long_help.txt:
```diff
@@ -0,0 +1,6 @@
+
+# Change the default timeout duration of the status command
+# Defaults to 10s if unspecified 
+$ ockam status --timeout 15s
```
I would just write

```
$ ockam status
```

The rest of the information will be added from the long_about help text.

Note that you need to include the code block, as we do [here](https://github.com/build-trust/ockam/blob/7ea6e4aaf5af1eb59ce7e3c1f44198d8ada42577/implementations/rust/ockam/ockam_command/src/vault/static/create/after_long_help.txt).

adrianbenavides commented:
> Please review the [long about](https://github.com/build-trust/ockam/pull/8070/commits/6199b9221777391633b080a036908421bdaff4d7#diff-e3ee3eff2828dd84252702e52b5b96e7ab705e2129a6278d767992926177fb1f) for status in my commit. Is that useful information or unnecessary duplication? Should anything in particular be elaborated on, and in what way?

I think I answered this in the review. Let me know if you have any further questions!

> When might it be useful or necessary to do a reset (asides maybe discarding after initially testing or trying things out)?

The reset command should be advertised as a dangerous / "use with caution" command really (we should display this in the command docs somehow), as it's used to remove all the local state permanently. Usually, the user will use more granular delete commands (e.g. `node delete`, `identity delete`, `space delete`, ...). But, yeah, as you said, the reset command is mostly used in examples, or while developing, to clean up things easily and start from scratch.

Wryhder commented:
> > Please review the [long about](https://github.com/build-trust/ockam/pull/8070/commits/6199b9221777391633b080a036908421bdaff4d7#diff-e3ee3eff2828dd84252702e52b5b96e7ab705e2129a6278d767992926177fb1f) for status in my commit. Is that useful information or unnecessary duplication? Should anything in particular be elaborated on, and in what way?
> 
> I think I answered this in the review. Let me know if you have any further questions!
> 
> > When might it be useful or necessary to do a reset (asides maybe discarding after initially testing or trying things out)?
> 
> The reset command should be advertised as a dangerous / "use with caution" command really (we should display this in the command docs somehow), as it's used to remove all the local state permanently. Usually, the user will use more granular delete commands (e.g. `node delete`, `identity delete`, `space delete`, ...). But, yeah, as you said, the reset command is mostly used in examples, or while developing, to clean up things easily and start from scratch.

@adrianbenavides Thanks a lot for your review! And apologies for the delay on this PR. I'll resume work on it within the next couple of days.

Wryhder commented:
@adrianbenavides I've made the suggested modifications. Please take a look again when you get the chance. Thank you!

Wryhder commented:
> The reset command should be advertised as a dangerous / "use with caution" command really (we should display this in the command docs somehow), as it's used to remove all the local state permanently. Usually, the user will use more granular delete commands (e.g. `node delete`, `identity delete`, `space delete`, ...). But, yeah, as you said, the reset command is mostly used in examples, or while developing, to clean up things easily and start from scratch.

@adrianbenavides What about a superscript-type tag like this (saying "unsafe" or similar), in addition to the warning-level notification in the doc itself?

![doc_notification_tag](https://github.com/user-attachments/assets/5ec64b6f-0c09-4d9d-a2e0-d46051e1df73)

adrianbenavides commented:
Awesome, thanks @Wryhder ! I'll take a look as soon as possible 🙏

adrianbenavides commented:
> @adrianbenavides What about a superscript-type tag like this (saying "unsafe" or similar), in addition to the warning-level notification in the doc itself?

That would be nice indeed. We do something similar with the "preview" tag ([here](https://github.com/build-trust/ockam/blob/a98f025e27d61876b919c9bc70c486a3f41d4710/implementations/rust/ockam/ockam_command/src/node/show.rs#L38) and [here](https://command.ockam.io/manual/ockam-node-stop.html)) if that's useful.

adrianbenavides commented on implementations/rust/ockam/ockam_command/src/status/static/after_long_help.txt:
````diff
@@ -0,0 +1,4 @@
+```sh
+# Display information about all of the system's Spaces, Identities, and Nodes.
````
I'd simply say `Display information about your Ockam instance and your Orchestrator resources`.

We plan to add more information to the output of this command, so it should be as general as possible, instead of naming the exact objects that it will show.

adrianbenavides commented on implementations/rust/ockam/ockam_command/src/status/static/long_about.txt:
```diff
@@ -0,0 +1,10 @@
+The `status` command displays the current status of your machine. This includes information such as
```
> The `status` command displays the current status of your machine.

I'd delete this sentence and start the next one with `This commands shows information such as...`

adrianbenavides commented on implementations/rust/ockam/ockam_command/src/status/static/long_about.txt:
```diff
@@ -0,0 +1,10 @@
+and their enrollment status with the Ockam Orchestrator. Additionally, it shows a quick overview of any
+available Nodes: their status, associated routes and protocols, listeners and services, and more.
+
+Note that you can change the default timeout of the status command with the `--timeout` argument:
+
+# Defaults to 10s if unspecified
+$ ockam status --timeout 15s
```
This is probably not useful for the average user. I'd delete it as well.

adrianbenavides commented on implementations/rust/ockam/ockam_command/src/reset/static/long_about.txt:
```diff
@@ -0,0 +1,16 @@
+-----------------------------------------------------------------------
+WARNING:
+
+The `reset` command is dangerous and should be used with caution as it will remove all local
+state PERMANENTLY. This behaviour can be useful for cleaning up your development environment
+and starting afresh from scratch. However, you'll typically want to use specific delete commands such as
+`node delete`, `identity delete`, and `space delete` for more granular control on deletes.
+
+-----------------------------------------------------------------------
```
Note that this must look good both in the `--help` command output and in the [docs page](https://docs.ockam.io/reference/command).

We could try removing the `----` lines and also removing any line break, otherwise it will look broken in the command.

adrianbenavides commented on implementations/rust/ockam/ockam_command/src/reset/static/long_about.txt:
```diff
@@ -0,0 +1,16 @@
+-----------------------------------------------------------------------
+
+By default, the `reset` command will only remove the local state (including Identities and Nodes).
+This is essentially the `database.sqlite3` file at `$OCKAM_HOME`. Spaces and Projects will only be deleted
```
> This is essentially the `database.sqlite3` file at `$OCKAM_HOME`.

This is too detailed. The state can also be stored in a postgres database and not necessarily under the OCKAM_HOME directory. I'd delete it.

adrianbenavides reviewed (commented):
@Wryhder I've left some comments to improve the output a little bit but it looks nice, thanks for taking the time with this!

One final recommendation: make sure you run `cargo clippy` and `cargo fmt` and squash all the commits into a single one once you have everything ready.

Wryhder commented:
> @Wryhder I've left some comments to improve the output a little bit but it looks nice, thanks for taking the time with this!
> 
> One final recommendation: make sure you run `cargo clippy` and `cargo fmt` and squash all the commits into a single one once you have everything ready.

@adrianbenavides I've made the modifications, run `cargo clippy` and `cargo fmt`, and squashed the commits. Please take a look when you get the chance. Thank you for your help with this!

Wryhder commented:
> > @adrianbenavides What about a superscript-type tag like this (saying "unsafe" or similar), in addition to the warning-level notification in the doc itself?
> 
> That would be nice indeed. We do something similar with the "preview" tag ([here](https://github.com/build-trust/ockam/blob/a98f025e27d61876b919c9bc70c486a3f41d4710/implementations/rust/ockam/ockam_command/src/node/show.rs#L38) and [here](https://command.ockam.io/manual/ockam-node-stop.html)) if that's useful.

@adrianbenavides This should be in a new PR, yes?

adrianbenavides commented on implementations/rust/ockam/ockam_command/src/reset/static/after_long_help.txt:
```diff
@@ -0,0 +1,13 @@
+# Removes orchestrator resources (i.e. Spaces and Projects)
+# in addition to deleting the local state.
+# This will prompt you for confirmation before proceeding.
+$ ockam reset -all
```
`-all` should be `--all`

adrianbenavides commented on implementations/rust/ockam/ockam_command/src/status/static/long_about.txt:
```diff
@@ -0,0 +1,3 @@
+This command shows information such as the current version of Ockam in use, the Ockam Orchestrator and Project versions, as well as your Spaces and the users associated with them.
+It also lists any cryptographic Identities stored in your vault and their enrollment status with the Ockam Orchestrator.
+Additionally, it shows a quick overview of any available Nodes: their status, associated routes and protocols, listeners and services, and more.
```
Each paragraph should be separatad by a new line, otherwise markdown will treat them as a single paragraph

adrianbenavides commented on implementations/rust/ockam/ockam_command/src/reset/static/after_long_help.txt:
````diff
@@ -0,0 +1,13 @@
+```sh
+# Removes the local Ockam configuration including all Identities and Nodes.
+# This will prompt you for confirmation before proceeding.
````
Can you remove this line? Since `--yes` is a global argument, I think it's not necessary to explain this here.

adrianbenavides commented on implementations/rust/ockam/ockam_command/src/reset/static/after_long_help.txt:
```diff
@@ -0,0 +1,13 @@
+
+# Removes orchestrator resources (i.e. Spaces and Projects)
+# in addition to deleting the local state.
+# This will prompt you for confirmation before proceeding.
```
Same as above

Wryhder replied:
Thank you @adrianbenavides! I've noted your changes. I'll go ahead to resolve the conversations now.

adrianbenavides reviewed (commented):
Hey @Wryhder just left some minor changes and we are good to merge!

adrianbenavides commented:
@Wryhder there are a couple of CI jobs failing:
- the commit linter is complaining about the `CLI` word being in uppercase. You can just amend it using `cli` instead.
- the editorconfig linter is complaining about a couple of missing newlines at the end of the files. Take a look [here](https://github.com/build-trust/ockam/actions/runs/11897770867/job/33157165328?pr=8070) for more details

Wryhder commented:
> @Wryhder there are a couple of CI jobs failing:
> 
> * the commit linter is complaining about the `CLI` word being in uppercase. You can just amend it using `cli` instead.
> * the editorconfig linter is complaining about a couple of missing newlines at the end of the files. Take a look [here](https://github.com/build-trust/ockam/actions/runs/11897770867/job/33157165328?pr=8070) for more details

@adrianbenavides  I added the final newlines and modified the commit message while squashing the commits. I've successfully changed the commit message but the newlines seem to have disappeared. I'm trying to figure out if it's my editor or what else.

Wryhder commented:
@adrianbenavides There are still a couple of CI jobs failing. One is about my commit not being signed and verified. Wanted to let you know I'll take a look by weekend.

Wryhder committed:
docs(rust): update cli documentation for the `status` and `reset` commands

Signed-off-by: Wryhder <jemimahsummers@yahoo.com>

Wryhder commented:
@adrianbenavides I think this is good to go now. Please confirm when you get the chance. Thanks very much for your time and help with this.

adrianbenavides commented:
Thanks @Wryhder for taking care of that! It all looks good now. The failing CI job is not related to this PR, and will get fixed on develop soon. We'll merge your PR after that is fixed. Thanks again for the time you put into this 🙏

adrianbenavides commented:
Hey @Wryhder ! I've created a PR [here](https://github.com/build-trust/ockam/pull/8707) with your commit fixing the conflicts for you. I've cherry-picked your commit, so you will get all the credit for your contribution.

Sorry again for the delay. We've been pretty busy working in the latest release and I didn't have the time to come back to this.

I'm closing this PR in favor of the other.
